app/javascript/mastodon/components/column_header.jsx

```diff
@@ -52,182 +50,193 @@ BackButton.propTypes = {
   onlyIcon: PropTypes.bool,
 };
 
-class ColumnHeader extends PureComponent {
-  static propTypes = {
-    identity: identityContextPropShape,
-    intl: PropTypes.object.isRequired,
-    title: PropTypes.node,
-    icon: PropTypes.string,
-    iconComponent: PropTypes.func,
-    active: PropTypes.bool,
-    multiColumn: PropTypes.bool,
-    extraButton: PropTypes.node,
-    showBackButton: PropTypes.bool,
-    children: PropTypes.node,
-    pinned: PropTypes.bool,
-    placeholder: PropTypes.bool,
-    onPin: PropTypes.func,
-    onMove: PropTypes.func,
-    onClick: PropTypes.func,
-    appendContent: PropTypes.node,
-    collapseIssues: PropTypes.bool,
-    ...WithRouterPropTypes,
-  };
-
-  state = {
-    collapsed: true,
-    animating: false,
-  };
-
-  handleToggleClick = (e) => {
+export const ColumnHeader = ({
+  title,
+  icon,
+  iconComponent,
+  active,
+  children,
+  pinned,
+  multiColumn,
+  extraButton,
+  showBackButton,
+  placeholder,
+  appendContent,
+  collapseIssues,
+  onClick,
+  onMove,
+  onPin,
+}) => {
+  const { formatMessage } = useIntl();
+  const { signedIn } = useIdentity();
+  const history = useAppHistory();
+  const [collapsed, setCollapsed] = useState(true);
+  const [animating, setAnimating] = useState(false);
+
+  const handleToggleClick = useCallback((e) => {
     e.stopPropagation();
-    this.setState({ collapsed: !this.state.collapsed, animating: true });
-  };
+    setCollapsed(!collapsed);
+    setAnimating(true);
+  }, [setCollapsed, setAnimating, collapsed]);
 
-  handleTitleClick = () => {
-    this.props.onClick?.();
-  };
+  const handleTitleClick = useCallback(() => {
+    onClick?.();
+  }, [onClick]);
 
-  handleMoveLeft = () => {
-    this.props.onMove(-1);
-  };
+  const handleMoveLeft = useCallback(() => {
+    onMove(-1);
+  }, [onMove]);
 
-  handleMoveRight = () => {
-    this.props.onMove(1);
-  };
+  const handleMoveRight = useCallback(() => {
+    onMove(1);
+  }, [onMove]);
 
-  handleTransitionEnd = () => {
-    this.setState({ animating: false });
-  };
+  const handleTransitionEnd = useCallback(() => {
+    setAnimating(false);
+  }, [setAnimating]);
 
-  handlePin = () => {
-    if (!this.props.pinned) {
-      this.props.history.replace('/');
+  const handlePin = useCallback(() => {
+    if (!pinned) {
+      history.replace('/');
     }
 
-    this.props.onPin();
-  };
+    onPin();
+  }, [history, pinned, onPin]);
 
-  render () {
-    const { title, icon, iconComponent, active, children, pinned, multiColumn, extraButton, showBackButton, intl: { formatMessage }, placeholder, appendContent, collapseIssues, history } = this.props;
-    const { collapsed, animating } = this.state;
+  const wrapperClassName = classNames('column-header__wrapper', {
+    'active': active,
```

Comment:
I think you can simply use `{ active }` here (and same for the other class names below?)